vignettes/synapser.Rmd

```diff
@@ -237,46 +238,104 @@ annotations<-synSetAnnotations(project, annotations)
 
-TODO example with Activity
+### Creating an activity object:
+```{r collapse=TRUE}
+act <- Activity(
+  name='clustering',
+  description='whizzy clustering',
+  used=c('syn1234','syn1235'),
+  executed='syn4567')
+```
+Here, syn1234 and syn1235 might be two types of measurements on a common set of samples. Some whizzy clustering code might be referred to by syn4567. The used and executed can reference entities in Synapse or URLs.
 
-```{r eval=F}
-?Activity
+Alternatively, you can build an activity up piecemeal:
+```{r collapse=TRUE}
+act <- Activity(name='clustering', description='whizzy clustering')
+act$used(c('syn12345', 'syn12346'))
+act$executed('syn4567')
 ```
 
-## Tables
-TODO
+### Storing entities with provenance
 
-Tables can be built up by adding sets of rows that follow a user-defined schema
-and queried using a SQL-like syntax.
+The activity can be passed in when storing an Entity to set the Entity’s provenance:
+
+```{r collapse=TRUE}
+project <- synGet(project$properties$id)
+project <- synStore(project, activity=act)
+```
+
+We’ve now recorded that 'project' is the output of syn4567 applied to the data stored in syn1234 and syn1235.
+
+### Recording data source
+
+The synStore() has shortcuts for specifying the used and executed lists directly. For example, when storing a data entity, it’s a good idea to record its source:
+
+```{r collapse=TRUE}
+project <- synStore(
+  project,
+  activityName='data-r-us',
+  activityDescription='downloaded from data-r-us',
+  used='http://data-r-us.com/excellent/data.xyz')
+```
+
+For more information about the activity object:
 
 ```{r eval=F}
-?Schema
-?Column
-?synGetColumns
-?synGetTableColumns
+?Activity
 ```
 
-## Wikis
+## <a name="Tables"></a>Tables
 
-TODO
+Tables can be built up by adding sets of rows that follow a user-defined schema
+and queried using a SQL-like syntax. Please visit the [Table vignettes](tables.html) for more information.
+
+## <a name="Wikis"></a>Wikis
 
 Wiki pages can be attached to an Synapse entity (i.e. project, folder, file, etc).
 Text and graphics can be composed in markdown and rendered in the web view of
 the object.
 
+Creating a Wiki
+```{r collapse=TRUE}
+project <- synGet(project$properties$id)
+content <- "
+# My Wiki Page
+
+Here is a description of my **fantastic** project!
+"
+
+wiki <- Wiki(owner=project, title='My Wiki Page', markdown=content)
+wiki <- synStore(wiki)
+```
+Updating a Wiki
+```{r collapse=TRUE}
+project <- synGet(project$properties$id)
+wiki <- synGetWiki(project)
+wiki.markdown <- "
+# My Wiki Page
+
+Here is a description of my **fantastic** project! Let's
+*emphasize* the important stuff.
+"
+
+wiki <- synStore(wiki)
+```
+
+For more information:
 ```{r eval=F}
-?synGetWiki
 ?Wiki
+?synGetWiki
 ```
 
-## Evaluations
+## <a name="Evaluations"></a>Evaluations
```

Comment:
would be good to have a round-trip demonstration of creating an evaluation (including setting up quota), creating a submission, retrieving submission, retrieving submission status, and updating submission annotations.